### DISCUSSION TITLE
Problem to run the RasterioSource on Google Colab

### DISCUSSION THREAD

**Discussion by marcosdavi1415** [azavea/raster-vision#1596] [Q&A]
Hi! I am facing a problem to run the first cell about RasterSource. This mensage is shown: "Unable to open EPSG support file gcs.csv.  Try setting the GDAL_DATA environment variable to point to the directory containing EPSG csv files". Could someone tell me how i can solve this problem?

**Comment by AdeelH** [comment]
**Note:** Edited the answer. Hadn't seen this was about Colab before.

This is how you're supposed to do it in Colab. Let me know if this works.

```python
import os
from subprocess import check_output

os.environ['GDAL_DATA'] = check_output('pip show rasterio | grep Location | awk \'{print $NF"/rasterio/gdal_data/"}\'', shell=True).decode().strip()
```

**Reply by marcosdavi1415** [reply]
Thanks. I tried to do the suggested, but the **gdal_data** folder is not showing up inside the **raster-vision** folder, only the rasterio folder exists. I found the path to **gdal** folder, but the problem remains when I change environment variable path.

**Reply by AdeelH** [reply]
Sorry, I missed that you are using Colab. This is how you're supposed to do it in Colab. Run this before importing rasterio.

```python
import os
from subprocess import check_output

os.environ['GDAL_DATA'] = check_output('pip show rasterio | grep Location | awk \'{print $NF"/rasterio/gdal_data/"}\'', shell=True).decode().strip()
```

**Reply by marcosdavi1415** [reply]
Thanks again. I did the suggested, but the problem continues. Can you run the colab cells?

**Reply by AdeelH** [reply]
Here is a working example: https://colab.research.google.com/drive/1qUl_6McbLJr9KjrhHnx_SWbSYkLPL2uW?usp=sharing

It turns out I needed to set an AWS-related env var as well. LMK if that works.

**Reply by marcosdavi1415** [reply]
The previous issue was resolved, but only when I run on the master branch. When I compile on the corresponding branch of the colab, which is the **cd-blog**, the previous problem remains. And some dependencies that are present only in the **cd-blog** branch are needed to run the other cells of the colab, . Would it be possible to solve the previous problem to use the respective **cd-blog** branch?

**Reply by AdeelH** [reply]
I did not realize you were running the Change Detection notebook.

For the RV branch used in that notebook, it seems like a kernel restart is required after setting the environment variables. I have updated the notebook to move the env var step above the reboot step: https://colab.research.google.com/drive/1Ex5xbAx4epzReCQDsqvZLoTpF6AtfnHr?usp=sharing

**Reply by marcosdavi1415** [reply]
Thanks a lot for the help. Now the problem is solved and I can run all the code.
